homeassistant/components/tplink/light.py

```diff
@@ -269,6 +285,19 @@ def supported_features(self):
         """Flag supported features."""
         return self._light_features.supported_features
 
+    def _get_valid_temperature_range(self):
+        """Return the device-specific white temperature range (in Kelvin).
+
+        :return: White temperature range in Kelvin (minimum, maximum)
+        """
+        model = self.smartbulb.sys_info[LIGHT_SYSINFO_MODEL]
+        for obj, temp_range in TPLINK_KELVIN.items():
+            if re.match(obj, model):
+                return temp_range
```

Comment:
`_get_valid_temperature_range` is called by `_get_light_features` which is called by `attempt_update` which is called by `async_update`.

Meaning this is constantly redetermined each time the light updates its state? Should this be determined once?

Reply:
I don't believe that's correct.  It will only set the features the first time.  See https://github.com/home-assistant/core/blob/dev/homeassistant/components/tplink/light.py#L249

Reply:
Thanks for clarifying that!